lib/ai/chat_test.go

```diff
@@ -51,7 +51,7 @@ func TestChat_PromptTokens(t *testing.T) {
 					Content: "Hello",
 				},
 			},
-			want: 721,
+			want: 610,
```

Comment:
The values here should not change. I think that because now the embedder is optional is disabled as a tool which is not correct. We still need to include the "get available node" tools in the prompt. We just need to use a different implementation for fetching the nodes.

Reply:
The test was previously instantiating a tool that would panic if invoked. As embedding tool was not used for the test it did not crash. I'm not sure what we want to test here except the tokens are counted, but IMO it is saner to remove broken tools from the test. If a test really depends on this tool, it should make it available and pass valid configuration.

Reply:
We'll have valid use-cases to enable/disable tools. e.g. Athena query building if the user doesn't have access to audit logs. I think a modular agent toolbox is a pattern we'll benefit from.

Reply:
I do understand that, but then you may end up with zero tools and very bad UX. I think that saving tokens + not confusing LLM are good arguments, but we should be smarter about when each tool is available. Can we for now still require the node embedded and fix the situation later (in a proper way)?

Reply:
I like the idea of separating the tools from the agent itself. I think we will need that anyways, as when I was working on the "in-context SSH Assist" I needed a different agent with a completely different set of tools.
I think we could even rename the `DefaultToolbox` to `ChatToolbox` as it contains all tools required by the chat. Then as you mentioned, other agents (like tests) can implement a different set of available tools.
@xacrimon It this change going to conflict with your access request changes? If yes, we can ship it after this and your PR are merged.

Reply:
Yep, im refactoring that area a little, sounds good.